### PR TITLE
[XML] Highlight files starting with DOCTYPE as XML

### PR DESCRIPTION
**Pull request by deathaxe**
Apply XML syntax to all files starting with `<!DOCTYPE >`, if it is not html. `<!DOCTYPE html>` is reserved for HTML syntax.

Sometimes the <?xml ...> prolog may be omitted in XML files which may also have an arbitary extension such as `*.ts` which is not associated to XML otherwise.

### CODE REVIEW

File: XML/XML.sublime-syntax [hunk 1]
```diff
@@ -13,11 +13,13 @@ file_extensions:
...
-    (?x)
-    ^(?:
-        <\?xml\s
-     |  \s*<([\w-]+):Envelope\s+xmlns:\1\s*=\s*"http://schemas.xmlsoap.org/soap/envelope/"\s*>
-     )
+    (?x:
+      ^(?:
+          <\?xml\s
+        | \s*<([\w-]+):Envelope\s+xmlns:\1\s*=\s*"http://schemas.xmlsoap.org/soap/envelope/"\s*>
+        | \s*<!DOCTYPE\s+(?!html\b)
```
Comment (Thom1729) [inline XML/XML.sublime-syntax]: `\b` should probably be `(?=[ \t\n\f>])`. See https://html.spec.whatwg.org/multipage/parsing.html#doctype-name-state.

Comment (deathaxe) [reply XML/XML.sublime-syntax]: Good catch.

Comment (deathaxe) [reply XML/XML.sublime-syntax]: PS.: If I recall correctly, nested positive lookaheads don't work within other lookaheads.

File: XML/XML.sublime-syntax [hunk 2]
```diff
@@ -17,7 +17,7 @@ first_line_match: |-
       ^(?:
           <\?xml\s
         | \s*<([\w-]+):Envelope\s+xmlns:\1\s*=\s*"http://schemas.xmlsoap.org/soap/envelope/"\s*>
-        | \s*<!DOCTYPE\s+(?!html\b)
+        | \s*<!DOCTYPE\s+(?!html[ \t\n\f>])
```
Comment (jfcherng) [inline XML/XML.sublime-syntax]: we need this to be case-insensitive?

**Comment by deathaxe** [comment]
Oups. Wrong commit title.

Otherwise, thanks!

**Pull request merged**